src/datatable/fread.py

```diff
@@ -280,12 +280,10 @@ def _resolve_source_cmd(self, cmd):
         proc = subprocess.Popen(cmd, shell=True,
                                 stdout=subprocess.PIPE,
                                 stderr=subprocess.PIPE)
-        ret = proc.wait()
         msgout, msgerr = proc.communicate()
-        if ret:
+        if msgerr:
```

Comment:
It is conceivable to have a process that terminates with an error, but doesn't write anything to the stderr. Thus, checking the return code is more robust. However, the return code can be queried after `.communicate()` using `.poll()` or `.returncode`.

Reply:
Yeah, let me also add a check for the return code.

Reply:
Try `dt.fread(cmd="false")`

Reply:
perhaps we can even add this as a test?

Reply:
Sure, but better use `cmd="exit 1"` or something like that, because on Windows `cmd="false"` will pass with `Command not found` error message.